File: Algorithm-Swift/Algorithm-Swift/Inflearn/32_선택정렬.swift
//
//  32_선택정렬.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/08/09.
//

import Foundation

func solution32() {
    let _ = Int(readLine()!)!
    var nums = readLine()!.split(separator: " ").map { Int($0)! }

    for i in 0..<nums.count - 1 {
        var min = nums[i]
        var temp = i

        for j in i..<nums.count {
            if min > nums[j] {
                min = nums[j]
                temp = j
            }
        }

        nums.swapAt(i, temp)
    }

    nums.forEach { print($0, terminator: " ") }
}

File: Algorithm-CPP/32_선택정렬.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;

    int arr[n];
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    for(int i = 0; i < n - 1; i++) {
        int min = arr[i];
        int temp = i;

        for(int j = i + 1; j < n; j++) {
            if (min > arr[j]) {
                min = arr[j];
                temp = j;
            }
        }

        arr[temp] = arr[i]; 
        arr[i] = min;
    }

    for(int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }

    return 0;
}